### PR TITLE
DlclassifierModel support  binary classification

### LINKED ISSUE #2703: Model.transform predict all result into 1.0
@qiuxin2012 Hello,qiuxin:
I have found a new porblems.when I use Model.transform,it predict all result into 1.0
    val testRank = dlModel.transform(testInput.coalesce(4))
my model is two hidden layers ,first one is 256, second is 128,I use Adam ,learning rate is 1e-3, I use glorot_uniform method to init, the activation I use relu ,and betch size 1024,I trained 10 epoch.
my dataInput is about 185 size Vectors about 7000000 lines .
My code is like this :
val hidden_1 = Dense(outputDim = hidden1Number, activation = activationMethod, init = getInitMethod, inputShape = Shape(inputNumber))
    val hidden_2 = Dense(outputDim = hidden2Number, init = getInitMethod, activation = activationMethod)
    val hidden_3 = Dense(outputDim = 1, init = getInitMethod, activation = "sigmoid")

    val model = Sequential()
      .add(hidden_1)
      .add(hidden_2)
      .add(hidden_3)

    val criterion = BCECriterion()
    val estimator = new DLClassifier(model, criterion, Array(inputNumber))
      .setFeaturesCol("scaledFeatures")
      .setLabelCol("label")
      .setBatchSize(batchSize)
      .setMaxEpoch(maxEpoch)
      .setOptimMethod(optimMethod)
      .setValidation(Trigger.everyEpoch, testInput, Array(new AUC()), 1280)
    val dlModel = estimator.fit(trainInput.coalesce(4))
    val testRank = dlModel.transform(testInput.coalesce(4))
    testRank.show(1000, false)

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

DlclassifierModel support  binary classification

## How was this patch tested?
unit test

## Related links or issues (optional)
fixed #2703

### CODE REVIEW

**Comment by qiuxin2012** [comment]
jenkins passed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dlframes/DLClassifier.scala [hunk 1]
```diff
@@ -72,7 +72,12 @@ class DLClassifierModel[T: ClassTag](
   )(implicit ev: TensorNumeric[T]) extends DLModel[T](model, featureSize) {
 
   protected override def outputToPrediction(output: Tensor[T]): Any = {
-    ev.toType[Double](output.max(1)._2.valueAt(1))
+    if (output.size().deep == Array(1).deep) {
```
[the remark this answers is not in the document]
Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/dlframes/DLClassifier.scala]: deep means element-wise compare

**Comment by zhangxiaoli73** [comment]
code LGTM
